source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -728,6 +729,12 @@ eagerly so that any operations waiting on `Connections <#connection>`_ can retry
 as soon as possible. The pool MUST NOT rely on WaitQueueTimeoutMS to clear
 requests from the WaitQueue.
 
+Additionally, a pool SHOULD force the next maintenance step to remove perished 
+connections. The maintenance logic SHOULD ensure that prune step is aware about 
+changed pool generation. As part of this request, a pool SHOULD inform maintenance
+thread whether "in use" connections should be closed as well. A pool SHOULD be 
+informed about it via closeInUseConnections parameter in Clear method.
+
```

Comment:
Probably "a pool SHOULD force the next maintenance step..." needs to be clarified. Does this mean that the next step should be scheduled to run as soon as possible? 
Maybe something like: The next Background Thread Run SHOULD be scheduled as soon as possible. Next pruning iteration MUST close "in use" perished connections if requested by closeInUseConnections flag...

Reply:
done

Reply:
Maybe we should omit the last "requested by closeInUseConnections flag." part? As it might be interpreted as scheduling the next run sooner will be done only in `closeInUseConnections:true` case.

Reply:
If we do that, can we bump it to its own paragraph below this one? That way the `closeInUseConnections` and background thread scheduling parts will be in separate paragraphs.

Reply:
I did not notice the 
`A pool SHOULD allow immediate scheduling of the next background thread iteration after a clear is performed. ` sentence. I am fine with current wording.

nit: remove the dot from `connections.requested`

Reply:
done